pkg/server/plugin/credentialcomposer/v1.go

```diff
@@ -0,0 +1,384 @@
+
+type x509CAAttributesResponseV1 interface {
+	GetAttributes() *credentialcomposerv1.X509CAAttributes
+}
+
+func x509CAAttributesFromV1(pb *credentialcomposerv1.X509CAAttributes) (attributes X509CAAttributes, err error) {
+	attributes.Subject, err = subjectFromV1(pb.Subject)
+	if err != nil {
+		return X509CAAttributes{}, fmt.Errorf("subject: %w", err)
+	}
+	attributes.PolicyIdentifiers, err = policyIdentifiersFromV1(pb.PolicyIdentifiers)
+	if err != nil {
+		return X509CAAttributes{}, fmt.Errorf("policy identifiers: %w", err)
+	}
+	attributes.ExtraExtensions, err = extraExtensionsFromV1(pb.ExtraExtensions)
+	if err != nil {
+		return X509CAAttributes{}, fmt.Errorf("extra extensions: %w", err)
+	}
+	return attributes, nil
+}
+
+type x509SVIDAttributesResponseV1 interface {
+	GetAttributes() *credentialcomposerv1.X509SVIDAttributes
+}
+
+func x509SVIDAttributesToV1(attributes X509SVIDAttributes) (*credentialcomposerv1.X509SVIDAttributes, error) {
+	subject, err := subjectToV1(attributes.Subject)
+	if err != nil {
+		return nil, err
+	}
+	return &credentialcomposerv1.X509SVIDAttributes{
+		Subject:         subject,
+		DnsSans:         attributes.DNSNames,
+		ExtraExtensions: extraExtensionsToV1(attributes.ExtraExtensions),
+	}, nil
+}
+
+func x509SVIDAttributesFromV1(pb *credentialcomposerv1.X509SVIDAttributes) (attributes X509SVIDAttributes, err error) {
+	attributes.Subject, err = subjectFromV1(pb.Subject)
+	if err != nil {
+		return X509SVIDAttributes{}, fmt.Errorf("subject: %w", err)
+	}
+	attributes.DNSNames = pb.DnsSans
+	attributes.ExtraExtensions, err = extraExtensionsFromV1(pb.ExtraExtensions)
+	if err != nil {
+		return X509SVIDAttributes{}, fmt.Errorf("extra extensions: %w", err)
+	}
+	return attributes, nil
+}
+
+type jwtSVIDAttributesResponseV1 interface {
+	GetAttributes() *credentialcomposerv1.JWTSVIDAttributes
+}
+
+func jwtSVIDAttributesToV1(attributes JWTSVIDAttributes) (*credentialcomposerv1.JWTSVIDAttributes, error) {
+	if len(attributes.Claims) == 0 {
+		return nil, errors.New("invalid claims: cannot be empty")
+	}
+	claims, err := structpb.NewStruct(attributes.Claims)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode claims: %w", err)
+	}
+	return &credentialcomposerv1.JWTSVIDAttributes{
+		Claims: claims,
+	}, nil
+}
+
+func jwtSVIDAttributesFromV1(pb *credentialcomposerv1.JWTSVIDAttributes) JWTSVIDAttributes {
+	return JWTSVIDAttributes{
+		Claims: pb.Claims.AsMap(),
+	}
+}
+
+func subjectFromV1(in *credentialcomposerv1.DistinguishedName) (pkix.Name, error) {
+	if in == nil {
+		return pkix.Name{}, errors.New("cannot be empty")
+	}
+	extraNames, err := extraNamesFromV1(in.ExtraNames)
+	if err != nil {
+		return pkix.Name{}, fmt.Errorf("extra names: %w", err)
+	}
+	return pkix.Name{
+		Country:            in.Country,
+		Organization:       in.Organization,
+		OrganizationalUnit: in.OrganizationalUnit,
+		Locality:           in.Locality,
+		Province:           in.Province,
+		SerialNumber:       in.SerialNumber,
```

Comment:
StreetAddress and PostalCode are missing here. We are missing those in the DistinguishedName message in the proto.

Reply:
That was intentional on my part.... but I suppose we should add them. They are always supportable through the ExtraNames field though.